### PR TITLE
Add option which enables client IP logging

### PR DESCRIPTION
**Pull request by paked**
This was brought up in #184. I believe it is a useful feature, but that it should not be enabled by default (I don't care about my IP when developing locally).

### CODE REVIEW

**Comment by indexzero** [comment]
Could you ensure the tests pass?

**Comment by paked** [comment]
Sorry. Totally didnt realise :laughing:

**Comment by paked** [comment]
@indexzero would your like me to squash the extra commits here?

**Comment by paked** [comment]
@indexzero Anything I need to do to get this merged?

**Comment by indexzero** [comment]
Could you rebase it off master? That would help, but either way I'm on vacation right now in South America so I probably won't be able to look at this for another week, sorry.

**Comment by paked** [comment]
@indexzero I've just rebased it. No problem on the holiday, have a good time!

**Comment by eladmoshe** [comment]
+1

**Comment by roccomuso** [comment]
+1

**Comment by uglycustard** [comment]
+1
Is this coming soon?

**Comment by pps83** [comment]
hard to imagine how one could have added logging (and waste time for coloring user-agent strings) without adding remote address. I needed simple quick http server with ability to track who (e.g. remote ip) accesses what.

**Comment by paked** [comment]
:+1:

**Comment by gagan-bansal** [comment]
+1
@indexzero this is nice feature. Any specific reason for not merging. 
May be if @paked resolves the conflict then its very easy for @indexzero to merge .

File: README.md [hunk 1]
```diff
@@ -66,6 +66,8 @@ This will install `http-server` globally so that it may be run from the command
 
 `-U` or `--utc` Use UTC time format in log messages.
 
+`-ip` Enable client IP logging
```
Comment (thornjad) [inline README.md]: Because of POSIX conventions, we should make this `--ip`, otherwise it looks like `-i` and `-p` together.

**Comment by paked** [comment]
It's been 4 years since I originally opened this PR, I'm not interested in working on merging it myself.

Will close, but someone else should feel free to take up from where I left on.

**Pull request closed without merging**